src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxGeneratorExtensions.cs

```diff
@@ -46,10 +43,17 @@ public static SyntaxNode DeclarationExt(this SyntaxGenerator syntaxGenerator, IS
                 {
                     INamedTypeSymbol? baseType = method.ContainingType.BaseType;
                     // If the base type does not have default constructor.
-                    if (baseType != null && !baseType.Constructors.IsEmpty && baseType.Constructors.Where(c => c.Parameters.IsEmpty).Count() == 0)
+                    if (baseType != null && !baseType.Constructors.IsEmpty && baseType.Constructors.All(c => !c.Parameters.IsEmpty))
```

Comment:
We are missing a filter check here to only include constructors that are publicly accessible (i.e. filter out sealed ones): https://github.com/dotnet/arcade/blob/608f3c7aed68dec1314d75601c6f143747e78cc6/src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Methods.cs#L444-L445

Reply:
members are filter out before we generate SyntaxNode out of ISymbol. https://github.com/dotnet/sdk/blob/bfbcfe92e8df98f04dc5e1a8835b0270176f4b67/src/GenAPI/Microsoft.DotNet.GenAPI/CSharpFileBuilder.cs#L121

Reply:
The filtering makes sure that we only traverse symbols that are included. Here we are looking at the base type constructors of the included constructor. Nothing guarantees at that point that the base type constructors are included.

You can test that with a type that has a base type with an internal ctor. This should result in a compilation error because the internal ctor of the base type isn't emitted (assuming that the `IncludeVisibleOutsideOfAssembly` setting is false).

Reply:
Thanks, updated.